optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/main/UpgradeMain.java

```diff
@@ -43,6 +44,23 @@ public class UpgradeMain {
   }
 
   public static void main(final String... args) {
+
+    final boolean initSchemaEnabled =
+        ConfigurationServiceBuilder.createDefaultConfiguration()
+            .getElasticSearchConfiguration()
+            .getConnection()
+            .isInitSchemaEnabled();
+
+    final boolean clusterTaskCheckingEnabled =
+        ConfigurationServiceBuilder.createDefaultConfiguration()
+            .getElasticSearchConfiguration()
+            .getConnection()
+            .isClusterTaskCheckingEnabled();
+    if (!initSchemaEnabled || !clusterTaskCheckingEnabled) {
+      throw new UpgradeRuntimeException(
+          "Can't upgrade without enable cluster task checking " + "and init schema privileges");
+    }
```

Comment:
I actually think this should happen after the determination of `databaseType` on lines 47-51. The reason is that this check is only current important for Elasticsearch, whereas the current implementation will also fail upgrades for OpenSearch. I think this might be best suited after line 74, when some of the more common validation has finished, and just before the upgrade itself starts

Reply:
fixed